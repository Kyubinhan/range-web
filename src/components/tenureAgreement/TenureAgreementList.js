import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TenureAgreementListItem from './TenureAgreementListItem';

const propTypes = {
  tenureAgreements: PropTypes.array.isRequired,
}

const defaultProps = {
  tenureAgreements: []
}

class TenureAgreementList extends Component {
  state = {
    activeIndex: -1,
  }
  
  handleActiveRow = (index) => {
    const { activeIndex } = this.state
    const newIndex = activeIndex === index ? -1 : index;
    
    this.setState({ activeIndex: newIndex })
  }

  render() {
    const { tenureAgreements } = this.props; 
    const { activeIndex } = this.state;

    return (
      <div className="tenure-agreement-list">
        {tenureAgreements.map((tenureAgreement, index) => {
          return (
            <TenureAgreementListItem 
              key={index}
              index={index}
              isActive={activeIndex === index}
              tenureAgreement={tenureAgreement}
              onViewClicked={this.handleActiveRow}
            />
          );
        })}
      </div>
    );
  }
}

TenureAgreementList.propTypes = propTypes;
TenureAgreementList.defaultProps = defaultProps;
export default TenureAgreementList;