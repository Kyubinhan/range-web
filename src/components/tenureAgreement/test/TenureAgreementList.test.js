import React from 'react';
import { shallow } from 'enzyme';
import TenureAgreementList from '../TenureAgreementList';
import { getMockTenureAgreements } from './mockValues';

const props = {};
const setupProps = () => {
  props.tenureAgreements = getMockTenureAgreements(3); 
};

beforeEach(() => {
  setupProps();
});

describe('TenureAgreementList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TenureAgreementList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `TenureAgreementListItem` components correctly', () => {
    const wrapper = shallow(<TenureAgreementList {...props} />);
    const numberOfAgreements = props.tenureAgreements.length;
    expect(wrapper.find('TenureAgreementListItem').length).toEqual(numberOfAgreements);    
  });

  describe('Event handlers', () => {
    it('`handleActiveRow` set `activeIndex` state correctly', () => {
      const wrapper = shallow(<TenureAgreementList {...props} />);
      const mockIndex = 0;
  
      wrapper.instance().handleActiveRow(mockIndex);
      expect(wrapper.state().activeIndex).toEqual(mockIndex);
  
      wrapper.instance().handleActiveRow(mockIndex);
      expect(wrapper.state().activeIndex).toEqual(-1);
    });
  });
});
