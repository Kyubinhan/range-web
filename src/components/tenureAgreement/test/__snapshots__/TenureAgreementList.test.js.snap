// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenureAgreementList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TenureAgreementList
    tenureAgreements={
        Array [
            Object {
              "id": 1,
              "number": "0001",
              "rangeOfficer": Object {
                "name": "Obi-Wan Kenobi",
              },
              "region": "Victoria Ranch 1",
              "status": "Approved",
              "tenureHolder": Object {
                "name": "Luke Skywalker",
              },
            },
            Object {
              "id": 2,
              "number": "0002",
              "rangeOfficer": Object {
                "name": "Leia Organa",
              },
              "region": "Victoria Ranch 2",
              "status": "Approved",
              "tenureHolder": Object {
                "name": "Han Solo",
              },
            },
            Object {
              "id": 3,
              "number": "0003",
              "rangeOfficer": Object {
                "name": "Obi-Wan Kenobi",
              },
              "region": "Victoria Ranch 3",
              "status": "Approved",
              "tenureHolder": Object {
                "name": "Luke Skywalker",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Table
        as="table"
        selectable={true}
        singleLine={true}
>
        <TableHeader
                as="thead"
        >
                <TableRow
                        as="tr"
                        cellAs="td"
                >
                        <TableHeaderCell
                                as="th"
                        >
                                Range #
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Range Name
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Agreement Holder
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Staff Contact
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Status
                        </TableHeaderCell>
                </TableRow>
        </TableHeader>
        <TableBody
                as="tbody"
        >
                <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                Object {
                                        "id": 1,
                                        "number": "0001",
                                        "rangeOfficer": Object {
                                          "name": "Obi-Wan Kenobi",
                                        },
                                        "region": "Victoria Ranch 1",
                                        "status": "Approved",
                                        "tenureHolder": Object {
                                          "name": "Luke Skywalker",
                                        },
                                      }
                        }
                />
                <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                Object {
                                        "id": 2,
                                        "number": "0002",
                                        "rangeOfficer": Object {
                                          "name": "Leia Organa",
                                        },
                                        "region": "Victoria Ranch 2",
                                        "status": "Approved",
                                        "tenureHolder": Object {
                                          "name": "Han Solo",
                                        },
                                      }
                        }
                />
                <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                Object {
                                        "id": 3,
                                        "number": "0003",
                                        "rangeOfficer": Object {
                                          "name": "Obi-Wan Kenobi",
                                        },
                                        "region": "Victoria Ranch 3",
                                        "status": "Approved",
                                        "tenureHolder": Object {
                                          "name": "Luke Skywalker",
                                        },
                                      }
                        }
                />
        </TableBody>
</Table>,
      "className": "tenure-agreement-list",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "table",
        "children": Array [
          <TableHeader
            as="thead"
>
            <TableRow
                        as="tr"
                        cellAs="td"
            >
                        <TableHeaderCell
                                    as="th"
                        >
                                    Range #
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Range Name
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Agreement Holder
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Staff Contact
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Status
                        </TableHeaderCell>
            </TableRow>
</TableHeader>,
          <TableBody
            as="tbody"
>
            <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                    Object {
                                                "id": 1,
                                                "number": "0001",
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 1",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                        }
            />
            <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                    Object {
                                                "id": 2,
                                                "number": "0002",
                                                "rangeOfficer": Object {
                                                  "name": "Leia Organa",
                                                },
                                                "region": "Victoria Ranch 2",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Han Solo",
                                                },
                                              }
                        }
            />
            <withRouter(TenureAgreementListItem)
                        tenureAgreement={
                                    Object {
                                                "id": 3,
                                                "number": "0003",
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 3",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                        }
            />
</TableBody>,
        ],
        "selectable": true,
        "singleLine": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "thead",
            "children": <TableRow
              as="tr"
              cellAs="td"
>
              <TableHeaderCell
                            as="th"
              >
                            Range #
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Range Name
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Agreement Holder
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Staff Contact
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Status
              </TableHeaderCell>
</TableRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tr",
              "cellAs": "td",
              "children": Array [
                <TableHeaderCell
                  as="th"
>
                  Range #
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Range Name
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Agreement Holder
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Staff Contact
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Status
</TableHeaderCell>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Range #",
                },
                "ref": null,
                "rendered": "Range #",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Range Name",
                },
                "ref": null,
                "rendered": "Range Name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Agreement Holder",
                },
                "ref": null,
                "rendered": "Agreement Holder",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Staff Contact",
                },
                "ref": null,
                "rendered": "Staff Contact",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Status",
                },
                "ref": null,
                "rendered": "Status",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "tbody",
            "children": Array [
              <withRouter(TenureAgreementListItem)
                tenureAgreement={
                                Object {
                                                "id": 1,
                                                "number": "0001",
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 1",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                }
/>,
              <withRouter(TenureAgreementListItem)
                tenureAgreement={
                                Object {
                                                "id": 2,
                                                "number": "0002",
                                                "rangeOfficer": Object {
                                                  "name": "Leia Organa",
                                                },
                                                "region": "Victoria Ranch 2",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Han Solo",
                                                },
                                              }
                }
/>,
              <withRouter(TenureAgreementListItem)
                tenureAgreement={
                                Object {
                                                "id": 3,
                                                "number": "0003",
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 3",
                                                "status": "Approved",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "tenureAgreement": Object {
                  "id": 1,
                  "number": "0001",
                  "rangeOfficer": Object {
                    "name": "Obi-Wan Kenobi",
                  },
                  "region": "Victoria Ranch 1",
                  "status": "Approved",
                  "tenureHolder": Object {
                    "name": "Luke Skywalker",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "tenureAgreement": Object {
                  "id": 2,
                  "number": "0002",
                  "rangeOfficer": Object {
                    "name": "Leia Organa",
                  },
                  "region": "Victoria Ranch 2",
                  "status": "Approved",
                  "tenureHolder": Object {
                    "name": "Han Solo",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "tenureAgreement": Object {
                  "id": 3,
                  "number": "0003",
                  "rangeOfficer": Object {
                    "name": "Obi-Wan Kenobi",
                  },
                  "region": "Victoria Ranch 3",
                  "status": "Approved",
                  "tenureHolder": Object {
                    "name": "Luke Skywalker",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Table
          as="table"
          selectable={true}
          singleLine={true}
>
          <TableHeader
                    as="thead"
          >
                    <TableRow
                              as="tr"
                              cellAs="td"
                    >
                              <TableHeaderCell
                                        as="th"
                              >
                                        Range #
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Range Name
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Agreement Holder
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Staff Contact
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Status
                              </TableHeaderCell>
                    </TableRow>
          </TableHeader>
          <TableBody
                    as="tbody"
          >
                    <withRouter(TenureAgreementListItem)
                              tenureAgreement={
                                        Object {
                                                  "id": 1,
                                                  "number": "0001",
                                                  "rangeOfficer": Object {
                                                    "name": "Obi-Wan Kenobi",
                                                  },
                                                  "region": "Victoria Ranch 1",
                                                  "status": "Approved",
                                                  "tenureHolder": Object {
                                                    "name": "Luke Skywalker",
                                                  },
                                                }
                              }
                    />
                    <withRouter(TenureAgreementListItem)
                              tenureAgreement={
                                        Object {
                                                  "id": 2,
                                                  "number": "0002",
                                                  "rangeOfficer": Object {
                                                    "name": "Leia Organa",
                                                  },
                                                  "region": "Victoria Ranch 2",
                                                  "status": "Approved",
                                                  "tenureHolder": Object {
                                                    "name": "Han Solo",
                                                  },
                                                }
                              }
                    />
                    <withRouter(TenureAgreementListItem)
                              tenureAgreement={
                                        Object {
                                                  "id": 3,
                                                  "number": "0003",
                                                  "rangeOfficer": Object {
                                                    "name": "Obi-Wan Kenobi",
                                                  },
                                                  "region": "Victoria Ranch 3",
                                                  "status": "Approved",
                                                  "tenureHolder": Object {
                                                    "name": "Luke Skywalker",
                                                  },
                                                }
                              }
                    />
          </TableBody>
</Table>,
        "className": "tenure-agreement-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "table",
          "children": Array [
            <TableHeader
              as="thead"
>
              <TableRow
                            as="tr"
                            cellAs="td"
              >
                            <TableHeaderCell
                                          as="th"
                            >
                                          Range #
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Range Name
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Agreement Holder
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Staff Contact
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Status
                            </TableHeaderCell>
              </TableRow>
</TableHeader>,
            <TableBody
              as="tbody"
>
              <withRouter(TenureAgreementListItem)
                            tenureAgreement={
                                          Object {
                                                        "id": 1,
                                                        "number": "0001",
                                                        "rangeOfficer": Object {
                                                          "name": "Obi-Wan Kenobi",
                                                        },
                                                        "region": "Victoria Ranch 1",
                                                        "status": "Approved",
                                                        "tenureHolder": Object {
                                                          "name": "Luke Skywalker",
                                                        },
                                                      }
                            }
              />
              <withRouter(TenureAgreementListItem)
                            tenureAgreement={
                                          Object {
                                                        "id": 2,
                                                        "number": "0002",
                                                        "rangeOfficer": Object {
                                                          "name": "Leia Organa",
                                                        },
                                                        "region": "Victoria Ranch 2",
                                                        "status": "Approved",
                                                        "tenureHolder": Object {
                                                          "name": "Han Solo",
                                                        },
                                                      }
                            }
              />
              <withRouter(TenureAgreementListItem)
                            tenureAgreement={
                                          Object {
                                                        "id": 3,
                                                        "number": "0003",
                                                        "rangeOfficer": Object {
                                                          "name": "Obi-Wan Kenobi",
                                                        },
                                                        "region": "Victoria Ranch 3",
                                                        "status": "Approved",
                                                        "tenureHolder": Object {
                                                          "name": "Luke Skywalker",
                                                        },
                                                      }
                            }
              />
</TableBody>,
          ],
          "selectable": true,
          "singleLine": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "thead",
              "children": <TableRow
                as="tr"
                cellAs="td"
>
                <TableHeaderCell
                                as="th"
                >
                                Range #
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Range Name
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Agreement Holder
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Staff Contact
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Status
                </TableHeaderCell>
</TableRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tr",
                "cellAs": "td",
                "children": Array [
                  <TableHeaderCell
                    as="th"
>
                    Range #
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Range Name
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Agreement Holder
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Staff Contact
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Status
</TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Range #",
                  },
                  "ref": null,
                  "rendered": "Range #",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Range Name",
                  },
                  "ref": null,
                  "rendered": "Range Name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Agreement Holder",
                  },
                  "ref": null,
                  "rendered": "Agreement Holder",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Staff Contact",
                  },
                  "ref": null,
                  "rendered": "Staff Contact",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Status",
                  },
                  "ref": null,
                  "rendered": "Status",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tbody",
              "children": Array [
                <withRouter(TenureAgreementListItem)
                  tenureAgreement={
                                    Object {
                                                      "id": 1,
                                                      "number": "0001",
                                                      "rangeOfficer": Object {
                                                        "name": "Obi-Wan Kenobi",
                                                      },
                                                      "region": "Victoria Ranch 1",
                                                      "status": "Approved",
                                                      "tenureHolder": Object {
                                                        "name": "Luke Skywalker",
                                                      },
                                                    }
                  }
/>,
                <withRouter(TenureAgreementListItem)
                  tenureAgreement={
                                    Object {
                                                      "id": 2,
                                                      "number": "0002",
                                                      "rangeOfficer": Object {
                                                        "name": "Leia Organa",
                                                      },
                                                      "region": "Victoria Ranch 2",
                                                      "status": "Approved",
                                                      "tenureHolder": Object {
                                                        "name": "Han Solo",
                                                      },
                                                    }
                  }
/>,
                <withRouter(TenureAgreementListItem)
                  tenureAgreement={
                                    Object {
                                                      "id": 3,
                                                      "number": "0003",
                                                      "rangeOfficer": Object {
                                                        "name": "Obi-Wan Kenobi",
                                                      },
                                                      "region": "Victoria Ranch 3",
                                                      "status": "Approved",
                                                      "tenureHolder": Object {
                                                        "name": "Luke Skywalker",
                                                      },
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "tenureAgreement": Object {
                    "id": 1,
                    "number": "0001",
                    "rangeOfficer": Object {
                      "name": "Obi-Wan Kenobi",
                    },
                    "region": "Victoria Ranch 1",
                    "status": "Approved",
                    "tenureHolder": Object {
                      "name": "Luke Skywalker",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "tenureAgreement": Object {
                    "id": 2,
                    "number": "0002",
                    "rangeOfficer": Object {
                      "name": "Leia Organa",
                    },
                    "region": "Victoria Ranch 2",
                    "status": "Approved",
                    "tenureHolder": Object {
                      "name": "Han Solo",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "tenureAgreement": Object {
                    "id": 3,
                    "number": "0003",
                    "rangeOfficer": Object {
                      "name": "Obi-Wan Kenobi",
                    },
                    "region": "Victoria Ranch 3",
                    "status": "Approved",
                    "tenureHolder": Object {
                      "name": "Luke Skywalker",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
